utils/git.go

```diff
@@ -0,0 +1,138 @@
+package utils
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+	"path/filepath"
+	"time"
+
+	"github.com/google/go-github/github"
+	"github.com/qlik-oss/k-apis/config"
+	"golang.org/x/oauth2"
+
+	"gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/plumbing/object"
+	"gopkg.in/src-d/go-git.v4/plumbing/transport/http"
+)
+
+func CreatePR(cr *config.CRSpec, branch plumbing.ReferenceName) error {
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: cr.Git.AccessToken},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+	client := github.NewClient(tc)
+
+	newPR := &github.NewPullRequest{
+		Title:               github.String("k-apis PR"),
+		Head:                github.String(branch.String()),
+		Base:                github.String("master"),
+		Body:                github.String("auto generated pr"),
+		MaintainerCanModify: github.Bool(true),
+	}
+
+	_, _, err := client.PullRequests.Create(context.Background(), cr.Git.UserName, filepath.Base(cr.ManifestsRoot), newPR)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func CloneRepository(cr *config.CRSpec) (*git.Repository, error) {
```

Comment:
can there be a separate method to clone the repo by taking input (repoUrl, gitRef, username, password) ??
this method should work without username and password as well